### PR TITLE
Support Python 3.11

### PR DESCRIPTION
**Pull request by borzunov**
Python 3.11 (stable) has been released in Oct 2022 (https://www.python.org/downloads/release/python-3110/) and soon may become a standard version across many distros. We need to support it.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@ msgpack>=0.5.6
 sortedcontainers
 uvloop>=0.14.0
 grpcio-tools>=1.33.2
-protobuf>=3.12.2,<4.0.0
+protobuf==4.23.4
```
Comment (justheuristic) [inline requirements.txt:10]: Fixing protobuf version to __latest__ can cause potential conflicts with some older projects. We gotta figure out some reasonably old version where the code (still) works.

**Comment by justheuristic** [comment]
@borzunov i've added some tweaks to make py311 install properly. please take a look when u have tim

File: hivemind/averaging/matchmaking.py [hunk 2]
```diff
@@ -480,7 +480,8 @@ async def pop_next_leader(self) -> PeerID:
                 self.peer_id.to_bytes(),
             ):
                 await asyncio.wait(
-                    {self.update_finished.wait(), self.declared_expiration.wait()}, return_when=asyncio.FIRST_COMPLETED
+                    set(map(asyncio.create_task, (self.update_finished.wait(), self.declared_expiration.wait()))),
```
Comment (borzunov) [inline hivemind/averaging/matchmaking.py:483]:
```suggestion
                    {asyncio.create_task(self.update_finished.wait()), asyncio.create_task(self.declared_expiration.wait())},
```

That's much easier to read

File: hivemind/averaging/matchmaking.py [hunk 3]
```diff
@@ -511,7 +512,7 @@ async def _update_queue_periodically(self, key_manager: GroupKeyManager) -> None
             self.update_finished.set()
 
             await asyncio.wait(
-                {self.running.wait(), self.update_triggered.wait()},
+                set(map(asyncio.create_task, (self.running.wait(), self.update_triggered.wait()))),
```
Comment (borzunov) [inline hivemind/averaging/matchmaking.py:515]:
```suggestion
                {asyncio.create_task(self.running.wait()), asyncio.create_task(self.update_triggered.wait())},
```

That's much easier to read

Comment (justheuristic) [reply hivemind/averaging/matchmaking.py:515]: let's see if black agrees

File: requirements.txt (hunk 1 above)
Comment (borzunov) [reply requirements.txt:10]: Please use `protobuf>=X`

**Review by justheuristic — APPROVED** [review]
LGTM :smiling_imp:

**Comment by justheuristic** [comment]
tested: petals client v2.0.0.post3 works in colab with this version of hivemind and forced protobuf 4.x

**Pull request merged**